### PR TITLE
fix(AM-SonarqubeChanges): Refactor code

### PR DESCRIPTION
**Pull request by Rahul-D78**
Signed-off-by: Rahul-D78 <rahul.kumar@aveshasystems.com>

### CODE REVIEW

File: pkg/server/slicerouter_config.go [hunk 1]
```diff
@@ -282,17 +282,19 @@ func sliceRouterReconcileRoutingTable() error {
...
 // The next hop IP would be the IP address of the slice-gw that connects to the remote cluster.
 func sliceRouterInjectRoute(remoteSubnet string, nextHopIP string) error {
-	if time.Since(lastRoutingTableReconcileTime).Seconds() > routingTableReconcileInterval {
-		err := sliceRouterReconcileRoutingTable()
-		if err != nil {
-			logger.GlobalLogger.Errorf("Failed to reconcile routing table: %v", err)
-		}
-
-		lastRoutingTableReconcileTime = time.Now()
+	if time.Since(lastRoutingTableReconcileTime).Seconds() < routingTableReconcileInterval {
```
Comment (gourishkb) [inline pkg/server/slicerouter_config.go:285]: add a log message here before early return

Comment (Rahul-D78) [reply pkg/server/slicerouter_config.go:285]: Need some suggestions for writing a proper log message

Comment (gourishkb) [reply pkg/server/slicerouter_config.go:285]: Something like "Skipping reconcilation, haven't crossed the reconciliation interval yet."

**Review by gourishkb — APPROVED** [review]
Except for the log message comment, rest looks good to me. :+1:

**Review by rahulsawra98 — APPROVED** [review]

**Pull request merged**